File: Level 1/1.Prefix Sum & Frequency Array/problems/problem 8/sol.cpp
#include <bits/stdc++.h>

using namespace std;
#define int long long
#define endl '\n'

void solve(){
    int n , k;
    cin >> n >> k;
    string s;
    cin >> s;
    vector<int> freq(26);
    for(int i =0; i < s.size(); i++){
        freq[s[i]-'a']++;
    }
    vector<int> remove(26);
    for(int i =0; i< 26; i++){
        if(freq[i] <= k){
            remove[i] = freq[i];
            k -= freq[i];
        }
        else {
            remove[i] = k;
            break;
        }
    }
    string ans = "";
    for(auto ch : s){
        if(remove[ch-'a'] > 0){
            remove[ch-'a']--;
        }
        else {
            ans += ch;
        }
    }
    cout << ans << endl;
}

signed main() {

    ios::sync_with_stdio(false);
    cin.tie(0),cout.tie(NULL);

    int t = 1;
    // cin >> t;
    while(t--){
        solve();
    }
    return 0;
}

File: Level 1/1.Prefix Sum & Frequency Array/problems/Java/problem 8/sol.java
import java.util.*;
import java.io.*;

public class sol {

    static void solve(BufferedReader br, PrintWriter pw) throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());
        long n = Long.parseLong(st.nextToken());
        long k = Long.parseLong(st.nextToken());

        String s = br.readLine().trim();

        long[] freq = new long[26];
        for (char ch : s.toCharArray()) {
            freq[ch - 'a']++;
        }

        long[] remove = new long[26];
        for (int i = 0; i < 26; i++) {
            if (freq[i] <= k) {
                remove[i] = freq[i];
                k -= freq[i];
            } else {
                remove[i] = k;
                break;
            }
        }

        StringBuilder ans = new StringBuilder();
        for (char ch : s.toCharArray()) {
            if (remove[ch - 'a'] > 0) {
                remove[ch - 'a']--;
            } else {
                ans.append(ch);
            }
        }

        pw.println(ans.toString());
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));

        int t = 1;
        // t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            solve(br, pw);
        }

        pw.flush();
    }
}